src/app/shell/shell.component.ts

```diff
@@ -28,20 +28,21 @@ export class ShellComponent implements OnInit, OnDestroy {
   ngOnInit(): void {
     this.geolocationService.handleUserLocation((coords: Coords)=> {
       coords && this.geolocationService.locationDecode(coords, (result) => {
-        this.store.dispatch([new ConfirmCity(false), new SetCity({
+        this.geolocationService.confirmCity({
           district: " ",
           longitude: coords.lng,
           latitude: coords.lat,
           name: result.address.city || result.address.town || result.address.village || result.address.hamlet,
           region: " "
-        })]);
+        }, false)
       });
     });
 
     this.isParent$
-      .pipe(takeUntil(this.destroy$))
-      .subscribe((parent) => {
-      !!parent && this.store.dispatch([
+      .pipe(takeUntil(this.destroy$),
+        filter(parent => !!parent))
+      .subscribe(() => {
+       this.store.dispatch([
```

Comment:
I also see the you do not need the actual instance of parent, you just need to coordinate the action depending on the role. In this case this actions should be dispatched if the user as parent is received and we have his/her profile. I suggest to replace this subscription also to the role state subscription.